### PR TITLE
fix double or more nested for foreach/where etc.

### PR DESCRIPTION
**Pull request by xieofxie**
Close https://github.com/microsoft/botbuilder-dotnet/issues/3134

Hop I am doing it right..

### CODE REVIEW

**Comment by boydc2014** [comment]
Can you open an issue explaining the issue here to be fixed and describe in this PR what's the root cause and the approach to fix?

File: libraries/Microsoft.Bot.Expressions/BuiltInFunctions.cs [hunk 1]
```diff
@@ -1517,10 +1527,15 @@ private static Expression RewriteAccessor(Expression expression, string localVar
...
                 {
                     var str = expression.ToString();
-                    var prefix = "$global";
+                    var prefix = GlobalVariablePrefix;
                     if (str == localVarName || str.StartsWith(localVarName + "."))
                     {
-                        prefix = "$local";
+                        prefix = LocalVariablePrefix;
+                    }
+                    else if (str == LocalVariablePrefix)
```
Comment (boydc2014) [inline libraries/Microsoft.Bot.Expressions/BuiltInFunctions.cs]: As i go over the changes, besides the naming changes, here is only place that affect the behavior, right?

**Comment by xieofxie** [comment]
@boydc2014 here https://github.com/microsoft/botbuilder-dotnet/issues/3134 with my observation.

File: libraries/Microsoft.Bot.Expressions/BuiltInFunctions.cs (hunk 1 above)
Comment (xieofxie) [reply libraries/Microsoft.Bot.Expressions/BuiltInFunctions.cs]: Right

**Comment by boydc2014** [comment]
Thanks @xieofxie , i'm closing this now because this is not a general fix, i will open a PR to solve this by replace the rewriting mechanism with a "StackedMemory", will let me know when that is in.

**Pull request closed without merging**